src/NuGet.Clients/NuGet.CommandLine/SettingsCredentialProvider.cs

```diff
@@ -30,21 +31,20 @@ public SettingsCredentialProvider(Configuration.IPackageSourceProvider packageSo
 
         public ICredentials GetCredentials(Uri uri, IWebProxy proxy, CoreV2.NuGet.CredentialType credentialType, bool retrying)
         {
-            NetworkCredential credentials;
             // If we are retrying, the stored credentials must be invalid.
-            if (!retrying && (credentialType == CoreV2.NuGet.CredentialType.RequestCredentials) && TryGetCredentials(uri, out credentials))
+            if (!retrying && (credentialType == CoreV2.NuGet.CredentialType.RequestCredentials) && TryGetCredentials(uri, out var credentials, out var username))
```

Comment:
Do we really need to pass back the username? 
Why not just use it from the Credentials object?

Reply:
ICredentials doesn't directly expose the username

Reply:
ah, missed the fact that you're changing it to ICredentials.